test/data/dataset/test_dataset.py

```diff
@@ -383,17 +382,18 @@ def test_dataset_write_if_exists_fail_by_default(self):
     def test_dataset_write_if_exists_append(self):
         from cartoframes.examples import read_brooklyn_poverty
         df = read_brooklyn_poverty()
-        dataset = Dataset(df).upload(table_name=self.test_write_table, credentials=self.credentials)
-        self.test_write_table = dataset.table_name
+        Dataset(df).upload(table_name=self.test_write_table, credentials=self.credentials)
 
-        dataset = Dataset(df).upload(
-            if_exists=Dataset.APPEND, table_name=self.test_write_table, credentials=self.credentials)
+        truncate_query = 'TRUNCATE TABLE {}'.format(self.test_write_table)
```

Comment:
why the `TRUNCATE`?

If we want to test the append I find useful the table to be already populated.

Reply:
I found a unique index of the table complaining (about `cartodb_id`). I will revisit it to avoid the `truncate`

Reply:
I have changed the cartodb_id in the second upload